### PR TITLE
Edit link updates the existing hyperlink in place

### LINKED ISSUE #494: Editing an existing link via "Edit link" inserts a new link at the caret instead of updating it
### What happened:

When editing an existing hyperlink in Docs, the change is not applied to the
existing link. Instead, a brand-new link is inserted at the current click/caret
position, and the original link is left unchanged.

### What you expected to happen:

Clicking "Apply" in the "Edit link" flow should update the existing link
(its URL and/or display text) in place. No new link should be inserted.

### How to reproduce it (as minimally and precisely as possible):

1. In a Docs document, select some text and insert a link
   (e.g. select "example" → Insert link → `https://example.com` → Apply).
2. Click on the linked text.
3. Choose "Edit link".
4. Change the URL (e.g. to `https://example.org`) and click "Apply".
5. Observe: the original link on "example" is unchanged, and a new link is
   inserted at the clicked position instead of the existing link being updated.

### Anything else we need to know?:

If I select the existing link by dragging over it and then change the URL, it works as expected.

### Environment:

- Operating system: macOS 15.6.1(24G90)
- Browser and version: 150.0.7871.128 (arm64)

https://github.com/user-attachments/assets/f22b2160-2f31-4675-ba8b-0bd373ef0725

### PR DESCRIPTION
**Pull request by te-ing**
## Summary

`insertLink` gains a third branch for a collapsed caret inside an
existing link run: instead of inserting the URL as brand-new linked
text, it updates that run's `href` in place. `removeLink`'s
run-boundary walk is extracted into a shared `findLinkRunAt` helper
(`packages/docs/src/view/link-run.ts`), and the fix is applied to both
editors — Docs body (`editor.ts`) and Slides text boxes / table cells
(`text-box-editor.ts`).

## Why

Clicking inside an existing hyperlink and using "Edit link" → Apply
left the original link untouched and inserted duplicate linked text at
the caret, because `insertLink` only handled a non-empty selection and
a bare collapsed caret — there was no branch for a caret inside a link
run. Sharing `removeLink`'s boundary walk keeps the two code paths from
diverging on what counts as "the link at the caret".

## Linked Issues

Fixes #494

## Author checklist

- [x] I searched existing issues and PRs and confirmed this is not a duplicate.
- [x] I read every line of this diff and can explain why each change is necessary.
- [x] Changes follow the conventions in the touched packages; any deviations are explained in *Why* above.
- [x] If AI tools assisted with this PR, I noted where in *Notes for Reviewers* below.

## Verification

CI automatically posts a verification summary comment on this PR with
per-lane results for both `verify:self` and `verify:integration`.

- [ ] verify:self — CI comment shows ✅
- [ ] verify:integration — CI comment shows ✅ (or explicit skip reason below)

Skip reason (if applicable):

## Risk Assessment

- User-facing risk: Low. Only the "Edit link" → Apply path changes;
  selection-based insert and remove-link behavior are untouched. Two
  known edge cases fall back to the previous insert behavior: the
  `href` residue `normalizeInlines` leaves on a fully-emptied
  paragraph, and the boundary between two adjacent different links
  (the earlier run wins, matching what `getLinkAtCursor` shows in the
  popover — the link displayed is the link edited).
- Data/security risk: None. View-layer editing only; no schema, store,
  or URL-handling changes.
- Rollback plan: Revert the PR; no migration or data cleanup needed.

## Notes for Reviewers

- UI changes (screenshots/gifs if applicable): No visual change —
  behavior fix only (existing link's URL now updates in place).
- Follow-up work (if any): The two fallback edge cases above could be
  handled explicitly later if they turn out to matter in practice.
- Implemented with Claude Code assistance (commit co-authored).

## Summary by CodeRabbit

* **New Features**
  * “Edit link” now updates an existing hyperlink in place when the caret is inside a link run (including at edges), rather than inserting new URL text.
  * Link editing/removal are consistent across the Docs editor and Slides text boxes.
  * Improved accuracy when resolving link boundaries and adjacent linked text.

* **Bug Fixes**
  * Fixed caret targeting and link “href residue” edge cases so adjacent links are not shadowed.
  * Preserved caret position and undo behavior during in-place link changes.

* **Tests**
  * Added/expanded regression and unit coverage for in-place edit, remove, boundary, and empty-text scenarios.

* **Documentation**
  * Added updated Docs task notes for the “edit link” in-place behavior.

### CODE REVIEW

**Comment by hackerwins** [comment]
@claude review this.

File: packages/docs/src/view/link-run.ts [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+ * *different* links (end of A == start of B) the earlier run wins —
+ * the same tie-break as `getLinkAtCursor`, so the link a popover
+ * displays is the link that gets edited or removed.
+ *
+ * An empty-text inline cannot anchor a run: the `href` residue that
+ * `normalizeInlines` leaves on a fully-emptied paragraph is not an
+ * editable link, and a zero-width range would make `applyInlineStyle`
+ * a silent no-op. (Empty inlines are still crossed by the lo/hi walk
+ * when they sit inside a wider same-href run.)
+ */
```
[the remark this answers is not in the document]
Comment (hackerwins) [reply packages/docs/src/view/link-run.ts:30]: Fixed in 91b6b33.

On reachability: the specific shadow state `[{text:'', href:A}, {text:'abc', href:B}]` is **not** reachable through the store — `normalizeInlines` (block-helpers.ts:79) drops every empty-text inline, keeping one only as the sole fallback when the whole paragraph is emptied. So an empty href residue can never sit *before* a real adjacent link.

That said, the finding still stands: the docstring's parity claim was inaccurate, and the divergence is a real (if currently latent) footgun. So I applied your suggested `text.length > 0` guard to `getLinkAtCursor` in both `editor.ts` and `text-box-editor.ts`, which fully aligns it with `findLinkRunAt` and makes the docstring true.

One extra consequence worth flagging: the guard also changes the *sole* empty-residue case (a fully-emptied paragraph) — `getLinkAtCursor` now returns `undefined` there instead of the invisible residue href, matching `findLinkRunAt`. The popover shows "add link" rather than prefilling a href for text that isn't visible; Apply's insert-fallback is unchanged. Updated `text-box-edit-link-in-place.test.ts` to match, and added a regression test in `edit-link-in-place.test.ts` that injects the (otherwise unreachable) shadow state directly and asserts the real link wins.

**Review by hackerwins — APPROVED** [review]
Thanks for your contribution.
